src/schema/v2/viewingRoom.ts

```diff
@@ -0,0 +1,309 @@
+import {
+  GraphQLBoolean,
+  GraphQLFieldConfig,
+  GraphQLID,
+  GraphQLList,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLString,
+} from "graphql"
+import { InternalIDFields, NodeInterface } from "./object_identification"
+import { ResolverContext } from "types/graphql"
+import {
+  convertConnectionArgsToGravityArgs,
+  defineCustomLocale,
+} from "lib/helpers"
+import { createPageCursors } from "./fields/pagination"
+import { connectionFromArray } from "graphql-relay"
+import { artworkConnection } from "./artwork"
+import { pageable } from "relay-cursor-paging"
+import moment from "moment"
+import { PartnerType } from "./partner/partner"
+import { dateRange } from "lib/date"
+import { GravityARImageType } from "./GravityARImageType"
+// import PartnerArtworks from "./partner/partnerArtworks"
+// import { ViewingRoomSubsectionType } from "./viewingRoomSubsection"
+// import { ViewingRoomArtworkType } from "./viewingRoomArtwork"
+
+const LocaleEnViewingRoomRelativeShort = "en-viewing-room-relative-short"
+defineCustomLocale(LocaleEnViewingRoomRelativeShort, {
+  parentLocale: "en",
+  relativeTime: {
+    future: "soon",
+    s: "",
+    ss: "",
+    m: "",
+    mm: "",
+    h: "",
+    hh: "",
+    d: "",
+    dd: "",
+    M: "",
+    MM: "",
+    y: "",
+    yy: "",
+  },
+})
+
+const LocaleEnViewingRoomRelativeLong = "en-viewing-room-relative-long"
+defineCustomLocale(LocaleEnViewingRoomRelativeLong, {
+  parentLocale: "en",
+  relativeTime: {
+    s: "%d second",
+    ss: "%d seconds",
+    m: "%d minute",
+    mm: "%d minutes",
+    h: "%d hour",
+    hh: "%d hours",
+    d: "%d day",
+    dd: "%d days",
+    M: "%d month",
+    MM: "%d months",
+    y: "%d year",
+    yy: "%d years",
+  },
+})
+
+export const ViewingRoomType = new GraphQLObjectType<any, ResolverContext>({
+  name: "ViewingRoom",
+  interfaces: () => {
+    return [NodeInterface]
+  },
+  fields: () => {
+    return {
+      ...InternalIDFields,
```

Comment:
If `id` wasn't being included in the stitched type, we shouldn't add it (yet) here - as that will cause a breaking change in Relay clients. When `id` shows up in the schema, they'll automatically add it to fragments - leading to complexity around trying to roll it all out. Weird gotcha I ran into before. Probably _only_ `internalID` needs to be included here to start.

Reply:
Oh I remember this issue! Glad you've noticed it, viewing rooms indeed do not expose `id` field